### PR TITLE
Metrics:Add rpc and p2p flow metrics and reconstruction some code

### PR DESCRIPTION
**Pull request by qizheng09**
## Description

What is the purpose of the change?

This pull request add prometheus metrics for xchain server, including `rpcFlowIn`, `rpcFlowOut`, `p2pFlowIn`, `p2pFlowOut`.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Brief of your solution

Please describe your solution to solve the issue or feature request.

## How Has This Been Tested?

Install prometheus set the target to xchain node as following
```
  - job_name: 'server'
    static_configs:
    - targets: ['localhost:37201','localhost:37202']

```

### CODE REVIEW

File: core/p2p/p2pv2/server.go [hunk 1]
```diff
@@ -153,6 +161,12 @@ func (p *P2PServerV2) SendMessageWithResponse(ctx context.Context, msg *p2pPb.Xu
 	percentage := msgOpts.Percentage
 	p.log.Trace("Server SendMessage with response", "logid", msg.GetHeader().GetLogid(),
 		"msgType", msg.GetHeader().GetType(), "checksum", msg.GetHeader().GetDataCheckSum(), "peers", peers)
+
+	matricLabels := prom.Labels{
+		"bcname": msg.GetHeader().GetBcname(),
+		"type":   msg.GetHeader().GetType().String(),
+	}
+	matrics.DefaultServerMetrics.P2PFlowOut.With(matricLabels).Add(float64(proto.Size(msg)))
```
Comment (chunhui01) [inline core/p2p/p2pv2/server.go:169]: Can grcp interceptors be used to reduce duplicate code?  Examples: UnaryServerInterceptor.

File: core/server/server.go [hunk 2]
```diff
@@ -64,47 +74,27 @@ func (s *Server) PostTx(ctx context.Context, in *pb.TxStatus) (*pb.CommonReply,
 			s.mg.P2pSvr.SendMessage(context.Background(), msg, opts...)
 		}()
 	}
+	matricLabelsOut := prom.Labels{
+		"bcname": in.GetBcname(),
+		"type":   "PostTx",
+	}
+	matrics.DefaultServerMetrics.RPCFlowOut.With(matricLabelsOut).Add(float64(proto.Size(out)))
```
Comment (chunhui01) [inline core/server/server.go:81]: Can grcp interceptors be used to reduce duplicate code? Examples: UnaryServerInterceptor.

File: core/common/matrics/matrics_test.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package matrics
```
Comment (yucaowang) [inline core/common/matrics/matrics_test.go:1]: Do you means metrics?

File: core/p2p/p2pv1/server.go [hunk 4]
```diff
@@ -207,7 +215,11 @@ func (p *P2PServerV1) SendMessageWithResponse(ctx context.Context, msg *p2pPb.Xu
 			msg = p2p_base.Compress(msg)
 		}
 	}
-
+	matricLabels := prom.Labels{
+		"bcname": msg.GetHeader().GetBcname(),
+		"type":   msg.GetHeader().GetType().String(),
+	}
+	matrics.DefaultServerMetrics.P2PFlowOut.With(matricLabels).Add(float64(proto.Size(msg)))
```
Comment (yucaowang) [inline core/p2p/p2pv1/server.go:222]: Is the response data size of SendMessageWithResponse calculated?

File: core/server/server.go (hunk 2 above)
Comment (yucaowang) [reply core/server/server.go:81]: +1

**Review by yucaowang — APPROVED** [review]
LGTM

**Review by icexin — APPROVED** [review]

**Pull request merged**